common/auth.js

```diff
@@ -103,6 +105,18 @@ class Auth {
     Auth.#authConfig = options.authConfig;
     Auth.#caTrustCerts = ArkimeUtil.certificateFileToArray(options.caTrustFile);
 
+    if (Auth.#app && options.trustProxy !== undefined) {
+      if (options.trustProxy === true || options.trustProxy === 'true') {
+        Auth.#app.set('trust proxy', true);
+      } else if (options.trustProxy === false || options.trustProxy === 'false') {
+        Auth.#app.set('trust proxy', false);
+      } else if (!isNaN(options.trustProxy)) {
+        Auth.#app.set('trust proxy', parseInt(options.trustProxy));
+      } else {
+        Auth.#app.set('trust proxy', options.trustProxy);
+      }
+    }
+
```

Comment:
> Do you want to do deep validation of the data, is that your preference over something like this. (This would config to be JSON, so I guess this makes string config is slightly uglier):
> 
> ```js
> if (typeof options.trustProxy == string) {
>  Auth.#app.set('trust proxy', JSON.parse(options.trustProxy));
> } else {
>   Auth.#app.set('trust proxy', options.trustProxy);
> }
> ```

I don't really care, it could switch to that if it covers all the cases, would need to be tested.


> 
> I think your implementation is fine though. It looks like the `else` branch will allow a string through which lets you set a comma separated string: https://expressjs.com/en/guide/behind-proxies.html

yep, need to allow hosts/ips

> 
> This implementation will fallback to `Auth.#app.set('trust proxy', undefined)` if `options.trustProxy` is unset which I appears to be fine, but isn't documented to be equivalent to `false` by Express.

```if (Auth.#app && options.trustProxy !== undefined) {``` should stop that

Reply:
Ya, so `JSON.parse("localhost, host");` doesn't work, so unless you have another simplification I'll just leave it.

Have you had a chance to test to make sure it does what you want?